internal/resolver/dns/dns_resolver.go

```diff
@@ -86,14 +86,14 @@ var (
 	minDNSResRate = 30 * time.Second
 )
 
-var customAuthorityDialler = func(authority string) func(ctx context.Context, network, address string) (net.Conn, error) {
-	return func(ctx context.Context, network, address string) (net.Conn, error) {
+var addressDialer = func(authority string) func(ctx context.Context, network, address string) (net.Conn, error) {
```

Comment:
Nit: `s/authority/address/` since this is an "address dialer" not an "authority dialer" now.  And just return type `func(context.Context, string, string) (net.Conn, error) {` as there's no need to name the arguments to the function returned.